lib/view_component/preview.rb

```diff
@@ -86,6 +86,12 @@ def preview_example_template_path(example)
                 .sub(/\..*$/, "")
       end
 
+      # Returns the method body for the example from the preview file.
+      def preview_example_source(example)
+        source = self.instance_method(example.to_sym).source.split("\n")
+        source[1...(source.size - 1)].join("\n")
```

Comment:
What's going on with the split and join here? What is being removed?

Reply:
Given the following method: 

```
  def default
    render(
      ExampleComponent.new(
        title: "Example Component"
      )
    )
  end
```

`self.instance_method(example.to_sym)` returns the method above as a string. The split and join returns the string above without the first and last line so we remove the method signature and the closing `end`. The result of the method is being displayed with syntax highlighting to the user.

Reply:
I think I'd expect to include the method signature too

Reply:
I find it confusing to keep the `default` method definition around the `render`. The sample source code is usually meant to be ready to copy-paste, and having to remove the first and last lines manually makes it harder.

See for instance: https://primer.style/view-components/components/button